YunoHost/yunohost
# Implement global settings for https redirect

## The problem

Fix https://github.com/YunoHost/issues/issues/1024

- The fact that every web app has to add its own `if ($scheme = http) rewrite ^ https://$server_name$request_uri? permanent;` is madness...
- One may want to have an easy way to disable HTTPS redirection (for example in the context of yunohost behind a reverse proxy, such as our infrastructure)

## Solution

Have a global setting to control wether HTTPS redirect is enabled or not

(... and then remove the https redirect snippet in every nginx.conf in every app ...)


## PR Status

Tested and working on my side

## How to test

- `yunohost settings set security.nginx.redirect_to_https -v True`
- check that you're redirected to https, except for requests on .well-known stuff (assuming the target exists and the request is explicitly in http, not https)
- `yunohost settings set security.nginx.redirect_to_https -v False` removes the redirection (at least in the nginx conf, then there's the whole HSTS story etc)
